rust/src/lib.rs

```diff
@@ -3693,9 +3697,20 @@ pub unsafe extern "C" fn zcashlc_tor_lwd_conn_check_single_use_taddr(
             }
         }
 
-        Ok(found)
+        let res = match found {
+            None => AddressCheckResult::NotFound,
+            Some(addr) => {
+                let addr_str = addr.encode(&network);
+                AddressCheckResult::Found {
+                    address: CString::new(addr_str).unwrap().into_raw(),
```

Comment:
I generally prefer that these are implemented as constructors on the FFI type, so that the allocation and freeing of raw pointers are in the same file next to each other (making it easier to confirm that they are consistent, and also allowing reuse).